java-squid/src/main/java/org/sonar/java/se/ConstraintManager.java

```diff
@@ -62,14 +68,33 @@ public SymbolicValue createSymbolicValue(Tree syntaxNode) {
       case INSTANCE_OF:
         result = new SymbolicValue.InstanceOfSymbolicValue(counter);
         break;
+      case MEMBER_SELECT:
+        result = createIdentifierSymbolicValue(((MemberSelectExpressionTree) syntaxNode).identifier());
+        break;
+      case IDENTIFIER:
+        result = createIdentifierSymbolicValue((IdentifierTree) syntaxNode);
+        break;
       default:
-        result = symbolicValueFactory == null ? new SymbolicValue(counter) : symbolicValueFactory.createSymbolicValue(counter, syntaxNode);
-        symbolicValueFactory = null;
+        result = createDefaultSymbolicValue(syntaxNode);
     }
     counter++;
     return result;
   }
 
+  private SymbolicValue createIdentifierSymbolicValue(IdentifierTree identifier) {
+    if (BOOLEAN_CLASS_VALUES.contains(identifier.name()) && identifier.symbolType() != null && identifier.symbolType().is(BOOLEAN_CLASS)) {
```

Comment:
Why do you need to null test identifierTree.symbolType ? it should not happen. 
Moreover, we are looking for an identifier like "TRUE" or "FALSE" meaning it is not a type so we should get the type of the symbol this identifier refers to. 
As such we should have somethingy like `identifier.symbol().type().is("java.lang.Boolean")`  (no need for a constant if it is used only here.